src/core/service.c

```diff
@@ -2026,6 +2026,8 @@ static void service_enter_start(Service *s) {
                         goto fail;
                 }
 
+                service_set_state(s, SERVICE_START);
```

Comment:
I thought about this, and I think doing a fake transition is better. Changing the state has a bunch of effects, and more could be added in the future, and running SuccessAction= is just one of them. Implementation wise, this approach is also nicer, because we don't have to repeat any of the logic.

Reply:
hmm, ok. I must admit in the autostart logic we also need to transition post-stop → auto-restart, and do fake a dead/failed state in between, so i figure doing that here is OK too

Reply:
but still, this should have a comment, with an explanation

Reply:
Ah, OK. I just pushed, so I'll let the CI run and do this later.